lib/css_parser/rule_set.rb

```diff
@@ -357,6 +359,7 @@ def expand_dimensions_shorthand! # :nodoc:
         #
         # TODO: rgba, hsl, hsla
         value.gsub!(RE_COLOUR) { |c| c.gsub(/(\s*,\s*)/, ',') }
+        value.gsub!(RE_FUNCTIONS) { |c| c.gsub(/\s+/, WHITESPACE_REPLACEMENT) }
 
         matches = value.strip.split(/\s+/)
```

Comment:
would this be better ?
- keep the replacement close together
- replace less if all have the same values
- do not produce extra hash and string objects
```suggestion
        value.gsub!(RE_FUNCTIONS) { |c| c.gsub!(/\s+/, WHITESPACE_REPLACEMENT); c }
        matches = value.strip.split(/\s+/)
        matches.each { |c| c.gsub!(WHITESPACE_REPLACEMENT, ' '); c }
```


... and ideally wrap this up in a `preserve_whitespace_in_functions` method

Reply:
New version pushed, using `#split_value_preserving_function_whitespace` if that name is okay. How's this look?